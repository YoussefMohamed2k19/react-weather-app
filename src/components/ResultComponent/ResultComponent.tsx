import { inject, observer } from "mobx-react";
import Cards from "../Cards/Cards";
import styles from "./ResultComponent.module.scss";
import TimeZone from "../TimeZone/TimeZone";

interface IResultComponent {
  store?: any;
}

const ResultComponent = inject("store")(
  observer(({ store }: IResultComponent) => {
    const { result } = store;
    const { main, sys, timezone, wind, weather } = result;
    return (
      <div className={styles.cardParent}>
        <Cards
          weather={weather}
          data={main?.temp}
          label="Temperature"
          unit="Â° C"
        />
        <Cards data={wind?.speed} label="Wind" unit="km/h" icon="bi bi-wind" />
        <Cards
          data={main?.humidity}
          label="Humidity"
          unit="%"
          icon="bi bi-droplet"
        />
        <Cards
          data={main?.pressure}
          label="Pressure"
          unit="hPa"
          icon="bi bi-water"
        />

        <Cards
          data={<TimeZone input={sys?.sunrise} timezone={timezone} />}
          label="Sun Rise"
          icon="bi bi-sunrise"
        />
        <Cards
          data={<TimeZone input={sys?.sunset} timezone={timezone} />}
          label="Sun Set"
          icon="bi bi-sunset"
        />
      </div>
    );
  })
);

export default ResultComponent;
