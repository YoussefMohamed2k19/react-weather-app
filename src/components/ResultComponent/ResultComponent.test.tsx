import { render } from "@testing-library/react";
import ResultComponent from "./ResultComponent";

test("Testing the ResultComponent", () => {
  let store = {
    dataState: {
      result: {
        main: {
          temp: 10,
          humidity: 78,
          pressure: 1008,
        },
        sys: {
          sunrise: 1662523476,
          sunset: 1662523476,
        },
        timezone: 7200,
        wind: {
          speed: 7.8,
        },
        weather: [
          {
            id: 804,
            main: "Clouds",
            description: "overcast clouds",
            icon: "04d",
          },
        ],
      },
    },
  };
  const { container } = render(<ResultComponent store={store} />);
  expect(container).toMatchSnapshot();
});
