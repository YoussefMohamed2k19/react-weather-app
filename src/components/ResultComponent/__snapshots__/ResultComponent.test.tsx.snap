// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the ResultComponent 1`] = `
<div>
  <div
    class="cardParent"
  >
    <div
      class="itemCard"
    >
      <img
        alt=""
        src="http://openweathermap.org/img/wn/04d@2x.png"
      />
      <span
        class="fs-6 fw-bold"
      >
        10
         
        Â° C
      </span>
      <span
        class="fw-lighter text-secondary"
      >
        Temperature
      </span>
    </div>
    <div
      class="itemCard"
    >
      <i
        class="fs-2 bi bi-wind"
      />
      <span
        class="fs-6 fw-bold"
      >
        7.8
         
        km/h
      </span>
      <span
        class="fw-lighter text-secondary"
      >
        Wind
      </span>
    </div>
    <div
      class="itemCard"
    >
      <i
        class="fs-2 bi bi-droplet"
      />
      <span
        class="fs-6 fw-bold"
      >
        78
         
        %
      </span>
      <span
        class="fw-lighter text-secondary"
      >
        Humidity
      </span>
    </div>
    <div
      class="itemCard"
    >
      <i
        class="fs-2 bi bi-water"
      />
      <span
        class="fs-6 fw-bold"
      >
        1008
         
        hPa
      </span>
      <span
        class="fw-lighter text-secondary"
      >
        Pressure
      </span>
    </div>
    <div
      class="itemCard"
    >
      <i
        class="fs-2 bi bi-sunrise"
      />
      <span
        class="fs-6 fw-bold"
      >
        6 : 04 AM
         
      </span>
      <span
        class="fw-lighter text-secondary"
      >
        Sun Rise
      </span>
    </div>
    <div
      class="itemCard"
    >
      <i
        class="fs-2 bi bi-sunset"
      />
      <span
        class="fs-6 fw-bold"
      >
        6 : 04 AM
         
      </span>
      <span
        class="fw-lighter text-secondary"
      >
        Sun Set
      </span>
    </div>
  </div>
</div>
`;
